### PR TITLE
Add key events file

### PR DESCRIPTION
**Pull request by Laurae2**
PR proposal: adding a dictionary of key events to LightGBM in two forms:

* Table (hard to copy but readable)
* List (easy to copy)

This will allow everyone to keep track on what is happening on LightGBM master repository without having the upkeep of watching every Pull Request / Commit happening on this repository everyday, assuming we keep adding to that file the important PRs/commits as needed. Very useful for people who are randomly looking at this repository from time to time, or for newcomers who want to check out what happened in this repository in a glance.

Missing stuff:

* Key events which are not from PR (such as boost_from_average addition, etc.)
* Author names (should they be added? should we add a "Contributor" list with names when their name gets on the key event list + their most important contributions?)

Opinions @guolinke @chivee @wxchan ?

Quick example view:

image

### CODE REVIEW

File: docs/Key-Events.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+| 01/12/2016 | [16d1853](https://github.com/Microsoft/LightGBM/pull/94) | Release | Python support (Pull Request 94)
+| 11/11/2016 | [98be7e3](https://github.com/Microsoft/LightGBM/pull/67) | Feature | DART booster support (Pull Request 67)
+| 08/11/2016 | [785398a](https://github.com/Microsoft/LightGBM/pull/69) | Parameter | L1 Regularization, L2 Regularization, Minimum Gain to Split support (Pull Request 69)
+| 01/11/2016 | [aa796a8](https://github.com/Microsoft/LightGBM/pull/53) | Feature | Multiclass classification support (Pull Request 53)
+| 28/10/2016 | [c45d1d9](https://github.com/Microsoft/LightGBM/pull/42) | Feature | Feature Importance support (Pull Request 42)
+| 25/10/2016 | [a6a75fe](https://github.com/Microsoft/LightGBM/pull/35) | Parameter | Maximum Depth support (Pull Request 35)
+| 21/10/2016 | [7aaba32](https://github.com/Microsoft/LightGBM/pull/27) | Parameter | Early Stopping support (Pull Request 27)<br>With original ([7d4b6d4](https://github.com/Microsoft/LightGBM/pull/21) on 20/10/2017, Pull Request 21)
+| 24/10/2016 | [9fe0dea](https://github.com/Microsoft/LightGBM/pull/30) | Parameter | Leaf Index Prediction support (Pull Request 30)
```
Comment (wxchan) [inline docs/Key-Events.md]: not ordered by time here?

Comment (Laurae2) [inline docs/Key-Events.md]: fixed, found also more which were not ordered by date

File: docs/Key-Events.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+| 06/04/2017 | [bfb0217](https://github.com/Microsoft/LightGBM/pull/383) | Feature | Objective Transformations to the Objective (Pull Request 383)
+| 05/04/2017 | [d4c4d9a](https://github.com/Microsoft/LightGBM/pull/381) | Feature | Regression Speed Improvements (Pull Request 381)
+| 03/04/2017 | [b6c973a](https://github.com/Microsoft/LightGBM/pull/378) | Feature | Unloading Memory Fix support for R package (Pull Request 378)
+| 17/03/2017 | [06a915a](https://github.com/Microsoft/LightGBM/pull/347) | Feature | RDS support for R pacakge (Pull Request 347)
+| 10/03/2017 | [b7e5f07](https://github.com/Microsoft/LightGBM/pull/340) | Feature | Support Self-Contained R package (Pull Request 340)
+| 02/03/2017 | [4d6ff28](https://github.com/Microsoft/LightGBM/pull/330) | Feature | Feature Contribution Plot for R package (Pull Request 330)
+| 01/03/2017 | [166421e](https://github.com/Microsoft/LightGBM/pull/328) | Feature | Feature Importance Plot for R package (Pull Request 328)
+| 01/03/2017 | [083b5da](https://github.com/Microsoft/LightGBM/pull/314) | Parameter | Unbalanced Mutliclass support (Pull Request 314)
```
Comment (guolinke) [inline docs/Key-Events.md]: I think this is removed now.

Comment (Laurae2) [inline docs/Key-Events.md]: removed.

Do you have any important commits (not from PRs) you consider key events that should be added?

Comment (guolinke) [inline docs/Key-Events.md]: Thanks. I think this contains all importance commits already .

**Pull request merged**